apps/web/src/views/PositionManagers/components/AddLiquidity.tsx

```diff
@@ -0,0 +1,337 @@
+  amountA,
+  amountB,
+  contractAddress,
+  disabled,
+  onDone,
+}: AddLiquidityButtonProps) {
+  const { t } = useTranslation()
+  const { approvalState: approvalStateToken0, approveCallback: approveCallbackToken0 } = useApproveCallback(
+    amountA,
+    contractAddress,
+  )
+  const { approvalState: approvalStateToken1, approveCallback: approveCallbackToken1 } = useApproveCallback(
+    amountB,
+    contractAddress,
+  )
+  const positionManagerWrapperContract = usePositionManagerWrapperContract(contractAddress)
+  const { fetchWithCatchTxError, loading: pendingTx } = useCatchTxError()
+  const { toastSuccess } = useToast()
+
+  const mintThenDeposit = useCallback(async () => {
+    const receipt = await fetchWithCatchTxError(() =>
+      positionManagerWrapperContract.write.mintThenDeposit(
+        [amountA?.numerator ?? 0n, amountB?.numerator ?? 0n, '0x'],
+        {},
+      ),
+    )
+
+    if (receipt?.status) {
+      toastSuccess(
+        `${t('Staked')}!`,
+        <ToastDescriptionWithTx txHash={receipt.transactionHash}>
+          {t('Your funds have been staked in position manager.')}
+        </ToastDescriptionWithTx>,
+      )
+      onDone?.()
+    }
+  }, [amountA, amountB, positionManagerWrapperContract, t, toastSuccess, fetchWithCatchTxError, onDone])
+
+  const showAmountButtonA = useMemo(
+    () => amountA && approvalStateToken0 === ApprovalState.NOT_APPROVED,
+    [amountA, approvalStateToken0],
+  )
+  const showAmountButtonB = useMemo(
+    () => amountB && approvalStateToken1 === ApprovalState.NOT_APPROVED,
+    [amountB, approvalStateToken1],
+  )
+  const isConfirmButtonDisabled = useMemo(
+    () =>
+      disabled ||
+      (amountA && approvalStateToken0 !== ApprovalState.APPROVED) ||
+      (amountB && approvalStateToken1 !== ApprovalState.APPROVED),
+    [amountA, amountB, disabled, approvalStateToken0, approvalStateToken1],
+  )
+
+  return (
+    <>
+      {showAmountButtonA && (
+        <Button
+          width="100%"
+          variant="primary"
+          disabled={disabled || approvalStateToken0 === ApprovalState.PENDING}
+          onClick={approveCallbackToken0}
+        >
+          {t('Approve %symbol%', { symbol: amountA.currency.symbol })}
+        </Button>
+      )}
+      {showAmountButtonB && (
+        <Button
+          mt="0.5em"
+          width="100%"
+          variant="primary"
+          disabled={disabled || approvalStateToken1 === ApprovalState.PENDING}
+          onClick={approveCallbackToken1}
+        >
+          {t('Approve %symbol%', { symbol: amountB.currency.symbol })}
+        </Button>
+      )}
+      <Button
+        mt="0.5em"
+        width="100%"
+        variant="primary"
+        isLoading={pendingTx}
+        disabled={isConfirmButtonDisabled}
+        onClick={mintThenDeposit}
+      >
+        {t('Confirm')}
+      </Button>
+      <LinkExternal external margin="24.5px auto 0 auto" href="https://docs.pancakeswap.finance/">
```

Comment:
Update this before merge
```suggestion
      <LinkExternal external margin="24.5px auto 0 auto" href="https://docs.pancakeswap.finance/">
```